packages/web-components/src/components/va-search/va-search.tsx

```diff
@@ -0,0 +1,183 @@
+  shadow: true,
+})
+export class VaSearch {
+  @Element() el: HTMLElement;
+
+  /**
+   * Fires when the search input loses focus
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  inputBlurEvent: EventEmitter;
+
+  /**
+   * Fires when the search input loses focus after its value was changed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  inputChangeEvent: EventEmitter;
+
+  /**
+   * Fires when the search input gains focus
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  inputFocusEvent: EventEmitter;
+
+  /**
+   * Fires when the search input is focused and a key is pressed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  inputKeyDownEvent: EventEmitter;
+
+  /**
+   * Fires when the search button is clicked
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  buttonClickEvent: EventEmitter;
+
+  /**
+   * Fires when the search button gains focus
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  buttonFocusEvent: EventEmitter;
+
+  /**
+   * Fires when the search button is focused and a key is pressed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  buttonKeyDownEvent: EventEmitter;
+
+  /**
+   * Text displayed inside the search button
+   */
+  @Prop() buttonText?: string;
+
+  /**
+   * Hides the search button
+   */
+  @Prop() hideButton: boolean = false;
+
+  /**
+   * Initial value of input
+   */
+  @Prop() inputValue?: string;
+
+  /**
+   * Adds an aria-label attribute to the input and button
+   */
+  @Prop() label: string = 'Search';
+
+  // Input Event Handlers
+  private handleInputBlurEvent = (event: FocusEvent) => {
+    this.inputBlurEvent.emit(event);
+  };
+
+  private handleInputChangeEvent = (event: Event) => {
+    // const target = event.target as HTMLInputElement;
+    this.inputChangeEvent.emit(event);
+  };
+
+  private handleInputFocusEvent = (event: FocusEvent) => {
+    this.inputFocusEvent.emit(event);
+  };
+
+  private handleInputKeyDownEvent = (event: KeyboardEvent) => {
+    this.inputKeyDownEvent.emit(event);
+  };
+
+  // Button Event Handlers
+  private handleButtonClickEvent = (event: KeyboardEvent | MouseEvent) => {
+    this.buttonClickEvent.emit(event);
+  };
+
+  private handleButtonFocusEvent = (event: FocusEvent) => {
+    this.buttonFocusEvent.emit(event);
+  };
+
+  private handleButtonKeyDownEvent = (event: KeyboardEvent) => {
+    this.buttonKeyDownEvent.emit(event);
+  };
+
+  render() {
+    const {
+      buttonText,
+      handleButtonClickEvent,
+      handleButtonFocusEvent,
+      handleButtonKeyDownEvent,
+      handleInputBlurEvent,
+      handleInputChangeEvent,
+      handleInputFocusEvent,
+      handleInputKeyDownEvent,
+      hideButton,
+      inputValue,
+      label,
+    } = this;
+
+    return (
+      <Host>
+        <input
+          id="va-search-input"
+          aria-activedescendant={this.el.getAttribute('aria-activedescendant')}
```

Comment:
@andresriveratoro correct me if I am wrong here but I am assuming these are being added so if someone wants to do like:
```
<va-search aria-activedescendant="XYZ" />
```
It would get added to the input element in the light Dom because otherwise the aria element wouldn't be hit since it would exist in the shadow Dom. Is that correct?

Reply:
Your example of how to use it is correct. Previously, I had created props for this functionality, but it wasn't required because we can grab the value from the wrapper `<va-search>` and use it on the `input`.

This input element is in the shadow DOM. Are you asking if `aria-activedescendant` will work if the element specified exists in the light DOM?

Reply:
That's correct. In order to get around that we have to grab the value from the host and explicitly set it on whatever element requires it.